core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -4058,67 +4059,40 @@ private static class TranslateImplementor extends AbstractRexCallImplementor {
   }
 
   /**
-   * Implementation that calls a given {@link java.lang.reflect.Method}.
+   * Implementation that a {@link java.lang.reflect.Method}.
+   *
+   * <p>If there are several methods in the list, calls the first that has the
+   * right number of arguments.
    *
    * <p>When method is not static, a new instance of the required class is
    * created.
    */
   private static class ReflectiveImplementor extends AbstractRexCallImplementor {
-    protected final Method method;
+    protected final ImmutableList<? extends Method> methods;
 
-    ReflectiveImplementor(Method method, @Nullable NullPolicy nullPolicy) {
-      super("reflective_" + method.getName(), nullPolicy, false);
-      this.method = requireNonNull(method, "method");
+    ReflectiveImplementor(List<? extends Method> methods) {
+      super("reflective_" + methods.get(0).getName(), NullPolicy.STRICT, false);
```

Comment:
shall we check input methods list? otherwise it may cause NPE

Reply:
The NPE is a sufficient check. This constructor is only called by Calcite developers. If they get an NPE or IndexOutOfBoundsException they will know what to do.